### PR TITLE
RSARTE-176062: Remove Poco dependency for LibTCP

### PR DESCRIPTION
**Pull request by rajneshkumarmalik**
Removed Poco dependency from libTcp

### CODE REVIEW

File: LibTCPServer.emx [hunk 1]
```diff
@@ -459,27 +446,20 @@ srv->start();
...
-	    		portIndex = var_portIndex.convert&lt;int>();&#xD;
-	    }&#xD;
-	    catch (Poco::Exception e) {&#xD;
-	    	conn->replyStatus(ITCPConnection::Status::Error, &quot;Illegal portIndex: &quot; + e.message());&#xD;
-	    	return;&#xD;
-	    }&#xD;
-	    &#xD;
+		std::string portIndexString =  mp.find(&quot;portIndex&quot;)->second;&#xD;
+		if (!portIndexString.empty()) {&#xD;
+			portIndex = stoi(portIndexString);&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx]: stoi will throw an instance of `std::invalid_argument` in case of incorrect string content.
Please, add error handling.

File: LibTCPServer.emx [hunk 2]
```diff
@@ -347,32 +340,28 @@ srv->start();
...
-	}&#xD;
-	catch (JSONException e) {&#xD;
-		conn->replyStatus(ITCPConnection::Status::Error, &quot;Failed to parse JSON: &quot; + e.displayText());&#xD;
-		return;&#xD;
-	}&#xD;
-	Object::Ptr object = result.extract&lt;Object::Ptr>();&#xD;
-	Var var_cmd = object->get(&quot;command&quot;);&#xD;
-	std::string cmd = var_cmd.convert&lt;std::string>();&#xD;
+	std::map&lt;std::string, std::string> mp;&#xD;
+	RTJsonDecoding object(nullptr);&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx]: You are using nullptr here - are we limiting the support to C++ 11 and higher?

File: LibTCPServer.emx [hunk 3]
```diff
@@ -567,9 +547,7 @@ srv->start();
 					coding.put(replyData, replyType);	&#xD;
 					strResult.append(&quot;, \&quot;data\&quot; : &quot;);&#xD;
 					std::string data(buf.getString());&#xD;
-					std::stringstream out;&#xD;
-					Stringifier::formatString(data, out);&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx:571]: What did this `Stringifier::formatString(data, out);` do?

File: LibTCPServer.emx [hunk 4]
```diff
@@ -626,10 +603,7 @@ if (data != 0) {&#xD;
 	RTAsciiEncoding coding(&amp;buf);&#xD;
 	coding.put(data, type);	&#xD;
 	str.append(&quot;, \&quot;data\&quot; : &quot;);&#xD;
-	// Make sure the data is encoded as a valid JSON string&#xD;
-	std::stringstream json;&#xD;
-	Stringifier::formatString(buf.getString(), json);&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx:631]: The same question here: you have removed formatString. What was it used for?

File: LibTCPServer.emx [hunk 5]
```diff
@@ -670,65 +644,66 @@ str.append(&quot;}&quot;);&#xD;
...
+&#xD;
+	static const RTTimespec awhile( 1L, 0L );&#xD;
+	while(!ioM->wait( &amp;awhile)) {&#xD;
+		std::this_thread::sleep_for(std::chrono::seconds(1));&#xD;
+	}&#xD;
 &#xD;
 	// Read response, if any&#xD;
-	char buffer[1024];&#xD;
-	size_t n;&#xD;
+	char buffer[1024] = {0};&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx]: This is a concept from modern C++...
I do not think this is needed since you have `RTMemoryUtil::memset(buffer, 0, sizeof(buffer));` in the loop

File: LibTCPServer.emx [hunk 6]
```diff
@@ -670,65 +644,66 @@ str.append(&quot;}&quot;);&#xD;
...
-	StreamSocket socket(socketAddress);&#xD;
-	socket.sendBytes(str.c_str(), str.length());&#xD;
+	bool writeSuccess = false;&#xD;
 &#xD;
+	RTTcpSocket tcp;&#xD;
+	RTIOMonitor *ioM = new RTIOMonitor();&#xD;
+	tcp.registerWith(ioM);&#xD;
+	tcp.create();&#xD;
+	tcp.connect(static_cast&lt;const char*>(config.remoteHost.c_str()), static_cast&lt;int>(config.remotePort));&#xD;
+	tcp.write(static_cast&lt;const char*>(str.c_str()), static_cast&lt;int>(str.length()));&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx]: Why did you remove catching of exceptions? Are these functions not throwing anything in case of errors?
Please add error handling.

File: prebuilt_MSVS-17-x64.tcjs [hunk 7]
```diff
@@ -8,9 +8,5 @@ tc.inclusionPaths = [
     '$(RSA_RT_HOME)/tcpserver/include'
 ];
 tc.libraries = [
-    '$(RSA_RT_HOME)/tcpserver/libs/tcpServerLib_msvs17x64.lib',
```
Comment (elena-volkova-hcl) [inline prebuilt_MSVS-17-x64.tcjs:11]: prebuilt tcjs must include tcpServerLib library, isn't it?

File: tcpServerLib.tcjs [hunk 8]
```diff
@@ -9,15 +8,13 @@ tc.createTargetProject = true;
 tc.targetProject = '/LibTCPServer_target';
 tc.type = CppTransformType.Library;
 tc.compilationMakeType = topTC.compilationMakeType || MakeType.MS_nmake;
-tc.compileArguments = topTC.compileArguments || '$(DEBUG_TAG) /MDd';
-tc.inclusionPaths = topTC.inclusionPaths || [
-    pocoLoc + '\\Net\\include',
-    pocoLoc + '\\JSON\\include',
-    pocoLoc + '\\Foundation\\include'
+tc.compileArguments = topTC.compileArguments;
+tc.inclusionPaths = [
```
Comment (elena-volkova-hcl) [inline tcpServerLib.tcjs]: There is not need to define empty inclusionPaths. Remove it completely

File: tcpServerLib.tcjs [hunk 9]
```diff
@@ -9,15 +8,13 @@ tc.createTargetProject = true;
 tc.targetProject = '/LibTCPServer_target';
 tc.type = CppTransformType.Library;
 tc.compilationMakeType = topTC.compilationMakeType || MakeType.MS_nmake;
-tc.compileArguments = topTC.compileArguments || '$(DEBUG_TAG) /MDd';
```
Comment (elena-volkova-hcl) [inline tcpServerLib.tcjs:12]: Why did you remove these default arguments:
$(DEBUG_TAG) /MDd

File: LibTCPServer.emx [hunk 10]
```diff
@@ -670,65 +644,66 @@ str.append(&quot;}&quot;);&#xD;
...
+&#xD;
 	if (config.logToStdOut)&#xD;
 		std::cout &lt;&lt; &quot;Response for out message: &quot; &lt;&lt; received &lt;&lt; std::endl;&#xD;
 	handleResponseForOutMessage(received);&#xD;
-}&#xD;
-catch (Poco::Exception&amp; exc) {&#xD;
-	if (!writeSuccess)&#xD;
-		std::cerr &lt;&lt; &quot;Error: Failed to write to socket for &quot; &lt;&lt; config.remoteHost &lt;&lt; &quot;@&quot; &lt;&lt; config.remotePort &lt;&lt; &quot;: &quot; + exc.displayText() &lt;&lt; std::endl;&#xD;
 &#xD;
-	// We failed to read a response, which typically just means the other end didn't give any. So this is not to be treated as an error.&#xD;
```
Comment (elena-volkova-hcl) [inline LibTCPServer.emx:700]: check this comment

**Review by elena-volkova-hcl — CHANGES_REQUESTED** [review]
Please add exception handling and see more comments in the code.

File: LibTCPServer.emx (hunk 2 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: Since TargetRTS is having support for C++11, so we decided to get benefit of C++11 in libTcp.

File: LibTCPServer.emx (hunk 3 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx:571]: This function make sure that data is correctly formated in json format. In this function data is formatted correctly and we don't have such API in TargetRTS and so it is removed.

File: LibTCPServer.emx (hunk 4 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx:631]: This function make sure that data is correctly formated in json format. In this function data is formatted correctly and we don't have such API in TargetRTS and so it is removed.

File: LibTCPServer.emx (hunk 5 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: Yes, initialization to 0 can be removed.

File: LibTCPServer.emx (hunk 6 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: These TargetRTS API are not throwing exceptions but retuning failure case condition. So I will add error handling for these functions.

File: prebuilt_MSVS-17-x64.tcjs (hunk 7 above)
Comment (rajneshkumarmalik) [reply prebuilt_MSVS-17-x64.tcjs:11]: Yes, I will add it.

File: tcpServerLib.tcjs (hunk 8 above)
Comment (rajneshkumarmalik) [reply tcpServerLib.tcjs]: Ok, I will remove it.

File: tcpServerLib.tcjs (hunk 9 above)
Comment (rajneshkumarmalik) [reply tcpServerLib.tcjs:12]: it got deleted by mistake, i will add it.

File: LibTCPServer.emx (hunk 1 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: Exception can be use for error handling with stoi() function but we have to avoid exception implementation. So I am planning to use stringstream class here.

Comment (elena-volkova-hcl) [reply LibTCPServer.emx]: @rajneshkumarmalik Why are we avoiding exceptions now? We have been using them a lot before to catch errors from poco.

Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: In bugbash meeting Mattias told to avoid exceptions. BTW it's slow down the execution.
